app/models/pipeline_run.rb

```diff
@@ -252,6 +252,11 @@ def report_ready?
     output_states.find_by(output: REPORT_READY_OUTPUT).state == STATUS_LOADED
   end
 
+  def report_failed?
+    # If the report ready output was marked as STATUS_FAILED
+    output_states.find_by(output: REPORT_READY_OUTPUT).state == STATUS_FAILED
```

Comment:
Thanks for adding the check :) FYI I don’t think the way it is written right now would give the correct behavior when host filtering fails. The host-filtering output_state would be failed, but the report-ready output_state would still be unknown

Reply:
Thanks, defined some constants for clarity